### PR TITLE
Attr tracking

### LINKED ISSUE #132: Within-File Attribute Tracking
Feature to tag metrics with within-file attributes from xarray and geopandas is required for experimental analysis. Xarray attributes should be used and geopandas columns should be used as well.

### PR DESCRIPTION
**Pull request by fernando-aristizabal**
Closes #132 
- Implements within file attribute tracking.
- Tags raster attributes from candidate and benchmark to an attributes df along with agreement map.
- Uses attributes df schema
- Implements tests.
- Creates attribute tracking accessor method.
- Implements attribute tracking within continuous and categorical compare accessor methods.

### CODE REVIEW

**Comment by fernando-aristizabal** [comment]
Made some changes as apparently it wasn't passing linting. Great use of automation!

**Review by GregoryPetrochenkov-NOAA — CHANGES_REQUESTED** [review]

File: src/gval/accessors/gval_xarray.py [hunk 1]
```diff
@@ -138,11 +184,18 @@ def categorical_compare(
...
+        attribute_tracking: bool, default = False
+            Whether to return a dataframe with the attributes of the candidate and benchmark maps.
+        attribute_tracking_kwargs: Optional[Dict], default = None
+            Keyword arguments to pass to `gval.attribute_tracking()`.  This is only used if `attribute_tracking` is True. By default, agreement maps are used for attribute tracking but this can be set to None within this argument to override. See `gval.attribute_tracking` for more information.
 
         Returns
         -------
-        Union[xr.Dataset, xr.DataArray], DataFrame[Crosstab_df], DataFrame[Metrics_df]
-            Tuple with agreement map, cross-tabulation table, and metric table
+        Union[
```
[the remark this answers is not in the document]
Comment (fernando-aristizabal) [reply src/gval/accessors/gval_xarray.py:194]: Good catch.

File: src/gval/accessors/gval_xarray.py [hunk 2]
```diff
@@ -220,11 +293,18 @@ def continuous_compare(
...
+        attribute_tracking: bool, default = False
+            Whether to return a dataframe with the attributes of the candidate and benchmark maps.
+        attribute_tracking_kwargs: Optional[Dict], default = None
+            Keyword arguments to pass to `gval.attribute_tracking()`.  This is only used if `attribute_tracking` is True. By default, agreement maps are used for attribute tracking but this can be set to None within this argument to override. See `gval.attribute_tracking` for more information.
 
         Returns
         -------
-        Union[xr.Dataset, xr.DataArray], DataFrame[Metrics_df]
-            Tuple with agreement map and metric table.
+        Union[
```
[the remark this answers is not in the document]
Comment (fernando-aristizabal) [reply src/gval/accessors/gval_xarray.py:303]: Changed.

**Comment by fernando-aristizabal** [comment]
Ready to review @GregoryPetrochenkov-NOAA pending successful CI builds and tests.

**Review by GregoryPetrochenkov-NOAA — APPROVED** [review]

**Pull request merged**